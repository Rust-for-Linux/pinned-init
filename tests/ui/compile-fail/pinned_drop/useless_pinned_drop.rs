use pin_init::*;
use std::pin::Pin;

#[pin_data]
struct Foo {}

#[pinned_drop]
impl PinnedDrop for Foo {
    fn drop(self: Pin<&mut Self>) {}
}

fn main() {}
