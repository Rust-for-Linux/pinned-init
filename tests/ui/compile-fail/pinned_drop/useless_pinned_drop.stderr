error[E0119]: conflicting implementations of trait `UselessPinnedDropImpl_you_need_to_specify_PinnedDrop` for type `Foo`
 --> tests/ui/compile-fail/pinned_drop/useless_pinned_drop.rs:4:1
  |
4 | #[pin_data]
  | ^^^^^^^^^^^
  | |
  | first implementation here
  | conflicting implementation for `Foo`
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
